contrib/mac/app/Makefile

```diff
@@ -11,37 +11,45 @@ ifeq ($(JULIA_PKGDIR),)
 endif
 
 # X.Y.Z or X.Y.Z-hash
-DMG_VERSION_SUFFIX:=$(shell [ $$(git describe --tags --exact-match 2>/dev/null) ] && echo $(JULIA_VERSION) || echo $(JULIA_VERSION)-$(JULIA_COMMIT))
+JULIA_VERSION_OPT_COMMIT:=$(shell [ $$(git describe --tags --exact-match 2>/dev/null) ] && echo $(JULIA_VERSION) || echo $(JULIA_VERSION)-$(JULIA_COMMIT))
 
 # X.Y
-APP_VERSION_SUFFIX:=$(shell echo $(JULIA_VERSION) | grep -o '^[0-9]\+.[0-9]\+')
+JULIA_VERSION_MAJOR_MINOR:=$(shell echo $(JULIA_VERSION) | grep -o '^[0-9]\+.[0-9]\+')
 
-DMG_NAME:=Julia-$(DMG_VERSION_SUFFIX).dmg
-APP_NAME:=Julia-$(APP_VERSION_SUFFIX).app
-VOL_NAME:=Julia-$(DMG_VERSION_SUFFIX)
+DMG_NAME:=Julia-$(JULIA_VERSION_OPT_COMMIT).dmg
+APP_NAME:=Julia-$(JULIA_VERSION_MAJOR_MINOR).app
+VOL_NAME:=Julia-$(JULIA_VERSION_OPT_COMMIT)
 
 all: clean $(DMG_NAME)
 
 $(DMG_NAME): dmg/$(APP_NAME)
 	-cp -f julia.icns dmg/.VolumeIcon.icns
 	-ln -fs /Applications ./dmg/Applications
-	-chmod 755 ./dmg/$(APP_NAME)/Contents/MacOS/Julia
+	-chmod 755 $</Contents/MacOS/applet
 	@echo "We're going to chown the .app file to root:admin now, which requires sudo.  You may be asked for your password:"
-	sudo chown -R root:admin ./dmg/$(APP_NAME)
-	sudo hdiutil create $(DMG_NAME) -size 500m -ov -volname "$(VOL_NAME)" -imagekey zlib-level=9 -srcfolder dmg
-
-dmg/$(APP_NAME): julia.icns
-	make -C ../../.. binary-dist
-	tar zxf ../../../julia-*.tar.gz
-	mv julia-* julia
-	-mkdir -p ./julia/libexec ./julia/share
-	rm -f julia/lib/*.{a,la}
-	-mkdir dmg
-	platypus -a Julia -p /bin/bash -V $(JULIA_VERSION) -R -u "The Julia Project" -i julia.icns -Q julia.icns -o "None" -I org.julialang.julia -x -f julia script ./dmg/$(APP_NAME)
-	-codesign -s "AFB379C0B4CBD9DB9A762797FC2AB5460A2B0DBE" --deep ./dmg/$(APP_NAME)
+	sudo chown -R root:admin $<
+	sudo hdiutil create $@ -size 500m -ov -volname "$(VOL_NAME)" -imagekey zlib-level=9 -srcfolder dmg
+
+
+dmg/$(APP_NAME): startup.applescript julia.icns
+	-mkdir -p dmg
+	osacompile -o $@ startup.applescript
+	rm $@/Contents/Resources/applet.icns
+	cp julia.icns $@/Contents/Resources/
+	plutil -replace CFBundleName -string "Julia" $@/Contents/Info.plist
+	plutil -replace CFBundleIconFile -string "julia.icns" $@/Contents/Info.plist
+	plutil -insert  CFBundleVersion -string "$(JULIA_VERSION_OPT_COMMIT)" $@/Contents/Info.plist
+	plutil -insert  CFBundleShortVersionString -string "$(JULIA_VERSION)" $@/Contents/Info.plist
+	plutil -insert  CFBundleIdentifier -string "org.julialang.julia" $@/Contents/Info.plist
+	plutil -insert  NSHumanReadableCopyright -string "© 2016 The Julia Project" $@/Contents/Info.plist
+	-mkdir -p $@/Contents/Resources/julia
+	make -C $(JULIAHOME) binary-dist
+	tar zxf ../../../julia-*.tar.gz -C $@/Contents/Resources/julia --strip-components 1
```

Comment:
According to the updated [`Resource Rules`](https://developer.apple.com/library/content/technotes/tn2206/_index.html#//apple_ref/doc/uid/DTS40007919-CH1-TNTAG206), it looks like all files are signed, there is no distinction made between "Resources" and "Binaries" anymore, so the "security" side of this distinction should now be moot.

As far as adherence to protocol goes, [Apple states](https://developer.apple.com/library/content/technotes/tn2206/_index.html#//apple_ref/doc/uid/DTS40007919-CH1-TNTAG201):

> ...putting code into other places will cause it to be sealed as data (resource) files. This causes this code to be sealed twice; once in the nested code, and once in the outer signature. This wastes both signing and verification time and storage space. Also, this can break the outer signature of apps that use their own update mechanisms to replace nested code. If this nested code is being treated as resources, the outer signature doesn't know that this nested content is actually code.

> Always put code and data into their proper places. This applies to all signed code and is enforced by the code signing machinery regardless of how the code is distributed.

They don't threaten us with codesigning breaking if we don't follow the rules, but they do say that the Mac App Store may reject submissions that don't follow these rules without warning:

> Note: While strict compliance with these rules may not affect your app today, anything that doesn't meet these requirements note may be rejected by code signing verification (and the Mac App Store validator, in the case of Mac App Store apps) at any point in the future without notice.

All this being said, I'm not at all against moving things around, so if you have a better layout in mind I'd love to hear it.  We can move things around pretty flexibly now by setting make flags [such as these](https://github.com/JuliaLang/julia/blob/master/Make.inc#L176-L184) to place our binaries in the proper location, and our non-binaries in others.